### PR TITLE
[DOCS] Split functions docs

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
This PR partially closes #5396 and splits functions section into smaller file.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

**Review by chriseth — CHANGES_REQUESTED** [review]
This reverts a change starting in the line `When a function has multiple return types,`...

**Comment by ChrisChinchilla** [comment]
@chriseth Thought I'd captured everything. Reapplied now.

File: docs/contracts/functions.rst [hunk 1]
```diff
@@ -106,9 +106,8 @@ return variables and then using ``return;`` to leave the function.
 Returning Multiple Values
 -------------------------
 
-When a function has multiple return types, the statement ``return (v0, v1, ..., vn) can be used to return multiple values.
-vn)`` can return multiple values.  The number of components must be
-the same as the number of return types.
+When a function has multiple return types, the statement ``return (v0, v1, ..., vn)`` can be used to return multiple values.
+vn)`` can return multiple values.  The number of components must be	The number of components must be the same as the number of return types.
```
Comment (chriseth) [inline docs/contracts/functions.rst]: still there

**Comment by ChrisChinchilla** [comment]
@chriseth Took this from `develop` so should be correct now.

**Comment by chriseth** [comment]
Rebased and fixed the pragma.

**Review by chriseth — APPROVED** [review]

**Pull request merged**